setup1-contributor/src/commands/contribute.rs

```diff
@@ -876,7 +877,27 @@ fn read_keys<P: AsRef<Path>>(keys_path: P, passphrase: Option<SecretString>) ->
     Ok((aleo_seed, aleo_private_key))
 }
 
-pub async fn start_contributor(opts: ContributeOptions) {
+async fn request_coordinator_public_settings(coordinator_url: &Url) -> anyhow::Result<PublicSettings> {
+    let settings_endpoint_url = coordinator_url.join("/v1/contributor/settings")?;
+    let client = reqwest::Client::new();
+    let bytes = client.post(settings_endpoint_url).send().await?.bytes().await?;
+    PublicSettings::decode(&bytes.to_vec())
+        .map_err(|e| anyhow::anyhow!("Error decoding coordinator PublicSettings: {}", e))
+}
+
+pub async fn contribute_subcommand(opts: &ContributeOptions) -> anyhow::Result<()> {
+    let public_settings = request_coordinator_public_settings(&opts.api_url).await.map_err(|e| {
+        tracing::error!("Can't get the coordinator public settings, got error: {}", e);
+        e
```

Comment:
perhaps it might be better to add this context to the error that gets returned rather than use the tracing error? At least the error that gets returned can include a backtrace (on nightly).

Reply:
Do you mean like using only context or both context and logging?

Reply:
You could do both if you really wanted to, but won't the error get reported anyway at a higher level in the code?